firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalStore.java

```diff
@@ -381,17 +382,19 @@ public SnapshotVersion getLastRemoteSnapshotVersion() {
             MaybeDocument doc = entry.getValue();
             MaybeDocument existingDoc = existingDocs.get(key);
 
-            // If a document update isn't authoritative, make sure we don't
-            // apply an old document version to the remote cache. We make an
-            // exception for SnapshotVersion.MIN which can happen for
-            // manufactured events (e.g. in the case of a limbo document
-            // resolution failing).
+            // If a document update isn't authoritative, make sure we don't apply an old document
+            // version to the remote cache.
             if (existingDoc == null
-                || doc.getVersion().equals(SnapshotVersion.NONE)
                 || (authoritativeUpdates.contains(doc.getKey()) && !existingDoc.hasPendingWrites())
                 || doc.getVersion().compareTo(existingDoc.getVersion()) >= 0) {
               remoteDocuments.add(doc);
               changedDocs.put(key, doc);
+              // NoDocuments with SnapshotVersion.MIN are used in manufactured events (e.g. in the
```

Comment:
This structure make the comment look as if it applies to the block about authoritative updates. Better to put this in the next block (and pull the comment above into this one--above the actions taken).

Reply:
Done